src/sssom/parsers.py

```diff
@@ -247,22 +249,27 @@ def parse_sssom_rdf(
 
 
 def parse_sssom_json(
-    file_path: str,
-    prefix_map: ConverterHint = None,
-    meta: Optional[MetadataType] = None,
-    **kwargs
-    # mapping_predicates: Optional[List[str]] = None,
+    file_path: str, prefix_map: ConverterHint = None, meta: Optional[MetadataType] = None, **kwargs
 ) -> MappingSetDataFrame:
-    """Parse a TSV to a :class:`MappingSetDocument` to a  :class`MappingSetDataFrame`."""
+    """Parse a TSV to a :class:`MappingSetDocument` to a :class:`MappingSetDataFrame`."""
     raise_for_bad_path(file_path)
-    converter, meta = _get_prefix_map_and_metadata(prefix_map=prefix_map, meta=meta)
 
     with open(file_path) as json_file:
         jsondoc = json.load(json_file)
+
+    # Initialize meta if it's None
+    if meta is None:
+        meta = {}
+
+    # Update metadata with values from JSON document
+    meta_keys_to_update = [MAPPING_SET_ID, LICENSE]
+    meta.update({key: jsondoc[key] for key in meta_keys_to_update if key in jsondoc})
```

Comment:
Why is this step necessary? There are dozens of `meta` level metadata elements in the sssom file, like descriptions and comments. Shouldnt this line load _all_ the mapping set level metadata? https://github.com/mapping-commons/sssom-py/blob/4c480a5d4bc419ba81f496ee3aa1152ddb0b5589/src/sssom/parsers.py#L491C5-L491C16

Reply:
The `meta` returned without the extra lines in question does not take into account the metadata in the JSON file.

In this line:
```
converter_via_metadata, meta = _get_prefix_map_and_metadata(prefix_map=prefix_map, meta=meta)
```
The `_get_prefix_map_and_metadata()` returns just the default metadata. In spite of taking the `jsondoc` into consideration, the metadata from JSON is not really considered.

Proof: I commented out these lines and ran the test. Gave me this:
```
AssertionError: {'cre[41 chars]ps://creativecommons.org/publicdomain/zero/1.0[160 chars]her'} != {'cre[41 chars]ps://w3id.org/sssom/license/unspecified', 'map[169 chars]her'}
E             {'creator_id': ['orcid:1234'],
E           -  'license': 'https://creativecommons.org/publicdomain/zero/1.0/',
E           +  'license': 'https://w3id.org/sssom/license/unspecified',
E              'mapping_date': '2020-05-30',
E           -  'mapping_set_id': 'https://w3id.org/sssom/mapping/tests/data/basic.tsv',
E           +  'mapping_set_id': 'https://w3id.org/sssom/mappings/cd9c5c7c-fe7a-49bc-9cb6-91d89de30867',
E              'mapping_tool': 'https://github.com/cmungall/rdf_matcher'}
```

Reply:
It's getting lost somewhere. I'll make this change in the correct location and remove it from the current location.

Reply:
Ok dug deeper:
Below the line linked by you above, there is this:
```
_set_metadata_in_mapping_set(mapping_set, metadata=meta)
``` 

https://github.com/mapping-commons/sssom-py/blob/f4256047f75f8d0a286b7255f76ab884d16e049d/src/sssom/parsers.py#L471-L486

This line  causes the issue. As a rule in this function `metadata` trumps `mapping_set` if key exists in both objects.

I'll try to explain this here so please bear with me.

In the absence of a metadata path in the param list while running the CLI for `sssom parse` (JSON=>TSV step) , a default metadata is adopted. 

So when we run:
```
with tsv_path.open("w") as tfile:
                parse_file(
                    input_path=str(json_path),
                    input_format="json",
                    prefix_map_mode="merged",
                    output=tfile,
                )
```
the param `metadata_path` is None (absent). This eventually calls `parse_sssom_json()` which looks like this:

https://github.com/mapping-commons/sssom-py/blob/f4256047f75f8d0a286b7255f76ab884d16e049d/src/sssom/parsers.py#L249-L258


Note `metadata_path` == `meta` which will be `None`. This is part triggers: 

https://github.com/mapping-commons/sssom-py/blob/f4256047f75f8d0a286b7255f76ab884d16e049d/src/sssom/parsers.py#L304-L305

Now the default metadata is: 
```
meta = {'mapping_set_id': 'https://w3id.org/sssom/mappings/d105f78d-a95a-445c-879f-e2ed4dc86aa7', 'license': 'https://w3id.org/sssom/license/unspecified'}
```
and hence the ones in the `mapping_set` (which are correct) get lost.

Reply:
And to answer the original question, this code seems appropriate but instead of just 2 attributes from `MappingSet` I have included all 
```
[attr for attr in dir(MappingSet) if not attr.startswith("_")]
```

Reply:
Wow thanks for the detailed explanation. Can we solve this problem in our next 1:1? There is a lot to unpack here, and not enough time for me digg into everything. But your `meta` analysis seems spot on, and this suggests there is a mistake in the design `meta`. 

Without going into details, the fact that we want `meta` to trump whatever is in the JSON (or TSV, or any) file is in conflict with the fact that providing _no_ meta leads to adding default metadata. Default metadata should only be added _after_ we are sure the sssom file does not have that information in it.

Thanks for this thorough analysis, was a pleasure to read!

Reply:
Sure! Just to add here , I tried rectifying the rule: `metadata` Trump's `mapping_set` by only adding key-value pairs from `metadata` into `mapping_set` but that breaks other tests . So I think there is a deeper problem we need to deal with in this case.

Reply:
I will close this comment as I understand the problem now; will work it out now.